support/util/imagemetadata.go

```diff
@@ -39,32 +42,85 @@ type RegistryClientImageMetadataProvider struct{}
 // tag is referring to could have changed. Once a digest is obtained, the cache is checked so that
 // no further fetching occurs. Only if both cache lookups fail, the image metadata is fetched and
 // stored in the cache.
-func (*RegistryClientImageMetadataProvider) ImageMetadata(ctx context.Context, imageRef string, pullSecret []byte, imageContentSource []hyperv1.ImageContentSource) (*dockerv1client.DockerImageConfig, error) {
+func (r *RegistryClientImageMetadataProvider) ImageMetadata(ctx context.Context, imageRef string, pullSecret []byte) (*dockerv1client.DockerImageConfig, error) {
```

Comment:
how do you know this changes work at all?
can we please make sure we capture the work needed to cover all this with unit tests and e2e smoke testing?

Reply:
We've tested the first commit in Elsa's disconnected environment. The second commit is not working yet.

Ack on the unit tests on e2e. This PR is still in progress.